subsys/bluetooth/host/keys.c

```diff
@@ -52,13 +60,32 @@ struct bt_keys *bt_keys_get_addr(u8_t id, const bt_addr_le_t *addr)
 		}
 	}
 
+#if defined(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
+	if (first_free_slot == ARRAY_SIZE(key_pool)) {
+		struct bt_keys *oldest = &key_pool[0];
+
+		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+			struct bt_keys *current = &key_pool[i];
+			if (current->aging_counter < oldest->aging_counter) {
+				oldest = current;
```

Comment:
At least to me this looks much nicer now with the helper variables!

Reply:
Agree